neuralpp/symbolic/expression.py

```diff
@@ -0,0 +1,151 @@
+from typing import Any, List
+from abc import ABC, abstractmethod
+
+
+class Expression(ABC):
+    @abstractmethod
+    # 'Expression': forward reference for recursive type reference https://peps.python.org/pep-0484/#forward-references
```

Comment:
FYI: you can use `from __future__ import annotations` to [postpone the evaluation of annotations](https://stackoverflow.com/questions/61544854/from-future-import-annotations), which should allow you to use `Expression` in type hinting (without using the string)

Reply:
Good to know! Thanks!